src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -379,6 +399,87 @@ def is_guid(guid):
         return False
 
 
+def get_latest_extension_version(extension_name='connectedk8s'):
+    try:
+        import re
+        git_url = "https://raw.githubusercontent.com/Azure/azure-cli-extensions/master/src/{}/setup.py".format(extension_name)
+        response = requests.get(git_url, timeout=10)
+        if response.status_code != 200:
+            logger.info("Failed to fetch the latest version from '%s' with status code '%s' and reason '%s'",
+                        git_url, response.status_code, response.reason)
+            return None
+        for line in response.iter_lines():
+            txt = line.decode('utf-8', errors='ignore')
+            if txt.startswith('VERSION'):
+                match = re.search(r'VERSION = \'(.*)\'$', txt)
+                if match:
+                    return match.group(1)
+                else:
+                    match = re.search(r'VERSION = \"(.*)\"$', txt)
+                    if match:
+                        return match.group(1)
+        return None
+    except Exception as ex:  # pylint: disable=broad-except
+        logger.info("Failed to get the latest version from '%s'. %s", git_url, str(ex))
+        return None
+
+
+def get_existing_extension_version(extension_name='connectedk8s'):
+    from azure.cli.core.extension import get_extensions
+    extensions = get_extensions()
+    if extensions:
+        for ext in extensions:
+            if ext.name == extension_name:
+                return ext.version or 'Unknown'
+
+    return 'NotFound'
+
+
+def check_connectivity(url='https://azure.microsoft.com', max_retries=5, timeout=1):
+    import timeit
+    start = timeit.default_timer()
+    success = None
+    try:
+        with requests.Session() as s:
+            s.mount(url, requests.adapters.HTTPAdapter(max_retries=max_retries))
+            s.head(url, timeout=timeout)
+            success = True
+    except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as ex:
```

Comment:
If any other exception occurs, troubleshoot cmd will crash. Can we handle that too and give appropriate message

Reply:
This was same as per what azure-cli does. https://github.com/Azure/azure-cli/blob/b24817537d6f589439ac19178d0ab7672e420598/src/azure-cli-core/azure/cli/core/util.py#L786

But, yes, will add a generic exception beyond that too.